sdk/python/kfp/components/structures/components.json_schema.json

```diff
@@ -0,0 +1,402 @@
+{
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "$id": "http://kubeflow.org/pipelines/components.json_schema.json",
+
+  "$ref": "#/definitions/PipelineRunSpec",
+
+  "definitions": {
+    "PrimitiveTypes": {
+      "oneOf": [
+        {"type": "string"},
+        {"type": "integer"},
+        {"type": "number"},
+        {"type": "boolean"}
+      ]
+    },
+
+    "TypeType": {
+      "oneOf": [
+        {"type": "string"},
+        {"type": "array", "items": {"$ref": "#/definitions/TypeType"}},
+        {"type": "object", "additionalProperties": {"$ref": "#/definitions/TypeType"}}
+      ]
+    },
+
+    "InputSpec": {
+      "description": "Describes the component input specification",
+      "type": "object",
+      "properties": {
+        "name": {"type": "string"},
+        "type": {"$ref": "#/definitions/TypeType"},
+        "description": {"type": "string"},
+        "default": {"$ref": "#/definitions/PrimitiveTypes"},
+        "optional": {"type": "boolean", "default": false}
+      },
+      "additionalProperties": false
+    },
+
+    "OutputSpec": {
+      "description": "Describes the component output specification",
+      "type": "object",
+      "required": ["name"],
+      "properties": {
+        "name": {"type": "string"},
+        "type": {"$ref": "#/definitions/TypeType"},
+        "description": {"type": "string"}
+      },
+      "additionalProperties": false
+    },
+
+    "InputValuePlaceholder": {
+      "description": "Represents the command-line argument placeholder that will be replaced at run-time by the input argument value.",
+      "type": "object",
+      "required": ["inputValue"],
+      "properties": {
+        "inputValue" : {
+          "description": "Name of the input.",
+          "type": "string"
+        }
+      },
+      "additionalProperties": false
+    },
+
+    "InputPathPlaceholder": {
```

Comment:
Why do we need a distinction between input values and input paths?

Reply:
It's the same distinction as "parameters" vs. "artifacts".
You cannot pass megabyte-sized model through command-line by value. You need to write it into a file and pass a file path. Hence `inputPath`.